### PR TITLE
Add documentation on splitting and repeating simulations, and deprecation warning for ResetNetwork

### LINKED ISSUE #525: Define what ResetNetwork is expected to do
Resetting a simulation mid-flight is a complex task. The function `ResetNetwork` is meant to do this, but the implementation is incomplete.

We thus decided to investigate what cannot be done explicitly without the function (e.g. resetting the internal spike buffers in neurons) and add corresponding functions. Once we have these functions in place, we deprecate `ResetNetwork` for the next release and advise users to use explicit calls to SetStatus for resetting neurons and synapses and the new functions for clearing the buffers.

After some time we remove `ResetNetwork` from all code and examples.

### PR DESCRIPTION
**Pull request by heplesser**
This also adds a missing copyright header to `import_libs.py`.

### CODE REVIEW

**Comment by heplesser** [comment]
This addresses #525.

File: nestkernel/nestmodule.cpp [hunk 1]
```diff
@@ -723,6 +723,10 @@ NestModule::ResetKernelFunction::execute( SLIInterpreter* i ) const
    Name: ResetNetwork - Reset the dynamic state of the network.
    Synopsis: ResetNetwork -> -
    Description:
+
+   ResetNetwork is deprecated as of NEST 2.18 and will be removed in NEST3,
```
Comment (hakonsbm) [inline nestkernel/nestmodule.cpp]: There should probably be a space between `NEST` and `3` for consistency.

File: pynest/nest/lib/hl_api_connections.py [hunk 2]
```diff
@@ -246,9 +246,9 @@ def Connect(pre, post, conn_spec=None, syn_spec=None, model=None):
     if model is not None:
         deprecation_text = "".join([
             "The argument 'model' is there for backward compatibility with ",
-            "the old Connect function and will be removed in a future",
-            "version of NEST. Please change the name of the keyword argument ",
-            "from 'model' to 'syn_spec'. For details, see the documentation ",
+            "the old Connect function and will be removed NEST 3. ",
```
Comment (hakonsbm) [inline pynest/nest/lib/hl_api_connections.py]: Should be "removed *in* NEST 3".

File: pynest/nest/lib/hl_api_helper.py [hunk 3]
```diff
@@ -136,7 +135,7 @@ def deprecated(alt_func_name, text=None):
     Parameters
     ----------
     alt_func_name : str, optional
-        Name of the function to use instead
+        Name of the function to use instead, may be em
```
Comment (hakonsbm) [inline pynest/nest/lib/hl_api_helper.py]: Should it be "may be *empty*"?

File: nestkernel/nestmodule.cpp (hunk 1 above)
Comment (heplesser) [reply nestkernel/nestmodule.cpp]: Now changed to NEST 3.0 throughout.

File: pynest/nest/lib/hl_api_connections.py (hunk 2 above)
Comment (heplesser) [reply pynest/nest/lib/hl_api_connections.py]: Thanks for catching this!

File: pynest/nest/lib/hl_api_helper.py (hunk 3 above)
Comment (heplesser) [reply pynest/nest/lib/hl_api_helper.py]: Thanks!

File: doc/guides/running_simulations.rst [hunk 4]
```diff
@@ -119,3 +119,75 @@ threshold crossing during the interval. Please see the documentation on
...
+are in the delivery pipeline, but it does not, e.g., reset plastic synapses.
+We therefore **advise against using** ``ResetNetwork()``.
+
+The only reliable way to perform two simulations of a network from exactly
+the same starting point is to restart NEST or to call `ResetKernel()` and
+then to build the network anew. If your simulations are rather large and
+you are working on a computer with a job queueing system, it may be most
+efficient to submit individual jobs or a job array to smiulate network 
+instances in parallel; don't forget to use different 
+:doc:`random seeds <random_numbers>`!
```
Comment (jougs) [inline doc/guides/running_simulations.rst]: I would rather describe in detail how to do repeated simulations properly than explain in detail what is not possible currently. The third paragraph could become the first, maybe extended with a small code example, while the first and second could be compressed to a short last paragraph along these lines:

```rst
The ``ResetNetwork()`` function available in NEST 2 is incomplete in that it only resets the
state of neurons and devices to default values and deletes spikes that are in the delivery
pipeline. It does does not reset plastic synapses or delete spikes from the spike buffers of
neurons. We therefore will remove the function in NEST 3 and already now **advise against
using** ``ResetNetwork()``.
```

File: doc/guides/running_simulations.rst [hunk 5]
```diff
@@ -119,3 +119,75 @@ threshold crossing during the interval. Please see the documentation on
...
+   - If you do not use ``RunManager()``, you must call ``Prepare()``, 
+     ``Run()`` and ``Cleanup()`` in that order
+   - You can call ``Run()`` any number of times inside a ``RunManager()`` 
+     context or between ``Prepare()`` and ``Cleanup()`` calls
+   - Calling ``SetStatus()`` inside a ``RunManager()`` context or
+     between ``Prepare()`` and ``Cleanup()`` will **lead to unpredictable
+     results** 
+   - After calling ``Cleanup()``, you need to call ``Prepare()`` again before
+     calling ``Run()``
+
```
[the remark this answers is not in the document]
Comment (heplesser) [reply doc/guides/running_simulations.rst]: Done

File: pynest/nest/lib/hl_api_helper.py [hunk 6]
```diff
@@ -116,9 +117,7 @@ def show_deprecation_warning(func_name, alt_func_name=None, text=None):
             alt_func_name = 'Connect'
         if text is None:
             text = "{0} is deprecated and will be removed in a future \
-            version of NEST.\nPlease use {1} instead!\n\
-            For details, see\
-            http://www.nest-simulator.org/connection_management\
+            version of NEST.\nPlease use {1} instead!\
             ".format(func_name, alt_func_name)
```
[the remark this answers is not in the document]
Comment (heplesser) [reply pynest/nest/lib/hl_api_helper.py]: @jougs Could we turn this into a separate issue? I just don't have time right now to wrap my head around how to format the text in such a way that it avoids ugly continuation lines and explicit `\n`, stays within our 80 character bound and works with `get_wrapped_text()`, but would like to get the updated documentation out to the user.

[the remark this answers is not in the document]
Comment (jougs) [reply pynest/nest/lib/hl_api_helper.py]: Agreed. Can you at least create an issue for this?

File: doc/guides/running_simulations.rst (hunk 4 above)
Comment (heplesser) [reply doc/guides/running_simulations.rst]: Done.

File: pynest/nest/lib/hl_api_helper.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (heplesser) [reply pynest/nest/lib/hl_api_helper.py]: Done, #1178.

**Pull request merged**